ste/sender-pageBlob.go

```diff
@@ -220,17 +223,41 @@ func (s *pageBlobSenderBase) Prologue(ps common.PrologueState) (destinationModif
 		if err := blockBlobTier.Parse(string(s.destBlobTier)); err != nil { // i.e it's not block blob tier
 			// Set the latest service version from sdk as service version in the context.
 			ctxWithLatestServiceVersion := context.WithValue(s.jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
-			if _, err := s.destPageBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{}); err != nil {
-				if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
-					s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-					s2sAccessTierFailureLogStdout.Do(func() {
-						glcm := common.GetLifecycleMgr()
-						glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-					})
+
+			// First, let's check if the intended tier is a premium page blob tier.
+			// These aren't available on non-premium storage accounts, but we can still copy the data.
+
+			shouldSet := true
+			if premiumPageBlobTierRegex.MatchString(string(s.destBlobTier)) {
+				infoResp, err := s.destPageBlobURL.GetAccountInfo(ctxWithLatestServiceVersion)
```

Comment:
It would be preferrable to get the account information once for the source account, and once for the destination.  Rather than once for every blob.  THis may not be an issue on page blobs, but it will be an issue on block blobs. As I'll note in another comment. 
I think ideally we'd compute some kind of boolean flag saying whether the source and dest accounts have compatible tier types (maybe wrapped with a sync.Once) and then just drive most of the logic here off that flag.

Reply:
Because not all destination blobs will have the same type or tier in case of S2S preservation, we can't just run the logic once. We can however, just grab the account details once.